src/NuGet.Clients/NuGet.Credentials/CredentialResponse.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+
+namespace NuGet.Credentials
+{
+    public class CredentialResponse
+    {
+        public ICredentials Credentials { get; set; }
+        public CredentialStatus Status { get; set; }
+        public string ErrorMessage { get; set; }
+
```

Comment:
I feel it's better to
- make the 3 properties read-only;
- add a constructor that takes 3 parameters matching the properties. Inside the constructor, we can validate the inputs to make sure only a valid response can be constructed. E.g. throws an exception if Status == Success but Credentials is null.